server/src/weaverbird/backends/sql_translator/steps/convert.py

```diff
@@ -13,6 +13,36 @@
 )
 from weaverbird.pipeline.steps import ConvertStep
 
+PG_BOOLEAN_VALUES = ','.join(
+    [
+        f'\'{el}\''
+        for el in ['t', 'true', 'y', 'yes', 'on', '1', 'f', 'false', 'n', 'no', 'off', '0']
+    ]
+)
+
+
+def build_psql(retrieved_col_type, data_type, col):
+    if retrieved_col_type == 'TEXT' and data_type == 'boolean':
+        return f'CASE WHEN {col} IN ({PG_BOOLEAN_VALUES}) THEN CAST({col} AS {data_type}) ELSE FALSE END AS {col}'
+    if retrieved_col_type == 'TEXT' and data_type == 'integer':
+        return f'CAST(NULLIF(SPLIT_PART(REGEXP_REPLACE({col}, \'[^0-9.]*\', \'\'), \'.\', 1), \'\') AS {data_type}) AS {col}'
```

Comment:
What's going on there ? :sweat_smile:

Reply:
Maybe a suboptimal way of converting strings to integers

Reply:
It's to avoid such errors in case when input is not a string representation of a  number 
> psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type integer: "meh"

So it basically replicates snowflake's convert step behaviour.

Reply:
-> Add a comment to explicit why we need such regexp